File: app/src/main/java/soup/movie/ui/theater/edit/TheaterEditContract.java
package soup.movie.ui.theater.edit;

import android.support.annotation.NonNull;

import java.util.List;

import soup.movie.data.model.Theater;
import soup.movie.ui.BaseContract;

class TheaterEditContract {

    interface Presenter extends BaseContract.Presenter<View> {
        void onConfirmClicked(List<Theater> selectedTheaters);
    }

    interface View extends BaseContract.View {
        void render(@NonNull TheaterEditViewState viewState);
    }
}

File: app/src/main/java/soup/movie/ui/theater/edit/TheaterEditContract.kt
package soup.movie.ui.theater.edit

import soup.movie.data.model.Theater
import soup.movie.ui.LegacyBaseContract

interface TheaterEditContract {

    interface Presenter : LegacyBaseContract.Presenter<View> {

        fun onConfirmClicked()

        fun remove(theater: Theater)
    }

    interface View : LegacyBaseContract.View {

        fun render(viewState: TheaterEditContentViewState)

        fun render(viewState: TheaterEditFooterViewState)
    }
}
